src/plugins/intel_gna/src/gna_graph_compiler.cpp

```diff
@@ -2392,6 +2392,40 @@ void GNAGraphCompiler::CreateLayerPrimitive(CNNLayerPtr layer) {
     }
 }
 
+bool GNAGraphCompiler::should_use_scratch_allocation(const InferenceEngine::CNNLayerPtr& layer, bool fp32_mode) const {
+    auto is_non_functional_layer = [](CNNLayerPtr l) {
+        return LayerInfo(l).isNonFunctional();
+    };
+
+    auto next_layer = CNNNetCheckNextLayerSkipCertain(layer, 0, 0, true, is_non_functional_layer).first;
+    auto prev_layer = CNNNetPrevLayerSkipCertain(layer, 0, is_non_functional_layer, true);
+
+    // In FP32 mode, max pooling or activation function are implemented as separate operations, which cannot be
+    // integrated into convolution.
+    if (!fp32_mode && Limitations::get_instance()->has_internal_memory_buffers()) {
```

Comment:
you could avoid one nest in case

if (fp32_mode || !Limitations::get_instance()->has_internal_memory_buffers()) {
    return true;
}

Reply:
OK